src/manifolds/Sphere.jl

```diff
@@ -1,30 +1,43 @@
 @doc raw"""
-    Sphere{N} <: AbstractEmbeddedManifold{DefaultEmbeddingType}
+    Sphere{N,M,𝔽} <: AbstractEmbeddedManifold{DefaultEmbeddingType}
+
+The (unit) sphere manifold $𝕊^{m,n}$ is the set of all unit norm elements. The sphere is
+represented in the embedding, and currently supports both vectors and matrices, i.e.
+
+````math
+𝕊^{n,m} := \bigl\{ p \in 𝔽^{m,n}\ \big|\ \lVert p \rVert = 1 \bigr\}
+````
+
+where $𝔽\in\{ℝ,ℂ\}. For $m=1$ and $𝔽=ℝ$ this  simplifies to unit vectors in $ℝ^n$ and the
+more common notation is $𝕊^{n,m} ≜ 𝕊^{n-1}$.
 
-The unit sphere manifold $𝕊^n$ represented by $n+1$-Tuples, i.e. in by
-vectors in $ℝ^{n+1}$ of unit length. A sphere is an embedded manifold
-and inherits the inner product of its embedding $\mathbb R^{n+1}$.
 The tangent space at point p is given by
 
 ````math
-T_p𝕊^n := \bigl\{ X ∈ ℝ^{n+1} : ⟨p,X⟩ = 0 \bigr \},
+T_p𝕊^{n,m} := \bigl\{ X ∈ 𝔽^{n,m} : ⟨p,X⟩ = 0 \bigr \},
 ````
 
-where $⟨\cdot,\cdot⟩$ denotes the inner product in the embedding $\mathbb R^{n+1}$.
+where $𝔽\in\{ℝ,ℂ\} and $⟨\cdot,\cdot⟩$ denotes the inner product in the
+embedding $\mathbb 𝔽^{n+1}$.
 
 This manifold is modeled as an embedded manifold to the [`Euclidean`](@ref), i.e.
 several functions like the [`inner`](@ref) product and the [`zero_tangent_vector`](@ref)
 are inherited from the embedding.
 
 # Constructor
 
-    Sphere(n)
+    Sphere(n[, field=ℝ])
 
-Generate $𝕊^{n} ⊂ ℝ^{n+1}$.
+Generate $𝕊^{n+1,1} ⊂ ℝ^{n+1}$.
+
+    Sphere(n,m[, field=ℝ])
+
+Generate $𝕊^{n,m} ⊂ ℝ^{n,m}$.
 """
-struct Sphere{N} <: AbstractEmbeddedManifold{DefaultIsometricEmbeddingType} end
+struct Sphere{n,m,𝔽} <: AbstractEmbeddedManifold{DefaultIsometricEmbeddingType} end
 
-Sphere(n::Int) = Sphere{n}()
+Sphere(n::Int, field::AbstractNumbers=ℝ) = Sphere{n+1,1,field}()
```

Comment:
I think it would be more consistent to use a tuple here instead of two size arguments. We did it similarly for Euclidean. By the way, we could even have multi-index spheres then.

Reply:
Yes, I already discussed with Seth shortly yesterday, that this could also cover unit norm tensors. Maybe its better (for consistency) to introduce a different name for the general case, since otherwise the argument of the constructor is inconsistent (n - dim of the manifold for vectors, (n1,n2,n3,...) - dim of embedding otherwise; dim is then their product -1). So I will do a `GeneralizedSphere` for the tensors and use the subtype `Sphere` just for convenience then speaking of vectors. That's the plan.

Reply:
That's a good idea.